### PR TITLE
docs: remove leftover TODO and warning from #1003.

### PR DESCRIPTION
**Pull request by wanda-phi**

### CODE REVIEW

**Review by whitequark — APPROVED** [review]
Thanks! We do describe what `with m.Case():` does in the docs, right?

**Comment by whitequark** [comment]
Do you want to fix the `matches` doc in this PR?

**Comment by wanda-phi** [comment]
> Do you want to fix the `matches` doc in this PR?

Yes, gimme a moment

**Comment by wanda-phi** [comment]
There goes.

File: docs/guide.rst [hunk 1]
```diff
@@ -762,20 +762,16 @@ Match operator
...
 
-If the pattern is a :class:`str`, it is treated as a bit mask with "don't care" bits. After removing whitespace, each character of the pattern is compared to the bit of the value in the same position as the character. If the pattern character is ``'0'`` or ``'1'``, the comparison succeeds if the bit equals ``0`` or ``1`` correspondingly. If the pattern character is ``'-'``, the comparison always succeeds. Aside from spaces and tabs, which are ignored, no other characters are accepted.
+If the pattern is a :class:`str`, it is treated as a bit mask with "don't care" bits. After removing whitespace, each character of the pattern is compared to the bit of the value in the same position as the character (counting characters from the right, like in a binary integer literal). If the pattern character is ``'0'`` or ``'1'``, the comparison succeeds if the bit equals ``0`` or ``1`` correspondingly. If the pattern character is ``'-'``, the comparison always succeeds. Aside from spaces and tabs, which are ignored, no other characters are accepted.
```
Comment (whitequark) [inline docs/guide.rst:765]: this is ... confusing

Comment (mcclure) [reply docs/guide.rst:765]: Perhaps don't say right, saying "from lowest to highest significance"?

Comment (whitequark) [reply docs/guide.rst:765]: I mean, which bit is the MSB and which bit is the LSB is what this sentence is explaining, so saying that would be vacuous.

**Review by whitequark — APPROVED** [review]
Thanks!

**Pull request merged**